### DISCUSSION TITLE
makeObservable and Typescript protected and private methods and properties

### DISCUSSION THREAD

**Discussion by ivandotv** [mobxjs/mobx#2490] [Q&A] [answered]
I have protected methods, and I need to make them in to actions however, the typescript complains that they don't exist.
The code looks like this:
```js
    makeObservable(this, {
      play: action,
      stop: action,
      pause: action,
      resume: action,
      // @ts-ignore - protected methods
      disposePlayer: action,
      initPlayer: action
    })
```

**Comment by mweststrate** [comment, marked as answer]
See docs, a generic arg with private keys can be passed in

Op di 6 okt. 2020 23:34 schreef Ivan V. <notifications@github.com>:

**Reply by kubk** [reply]
@mweststrate Is this the correct way to fix the issue?

```js
class Test {
  constructor() {
    makeObservable<Test, 'disposePlayer' | 'anotherPrivateKey'>(this, {
      play: action,
      stop: action,
      pause: action,
      resume: action,
      disposePlayer: action,
      initPlayer: action
    })
  }
}
```

**Reply by kubk** [reply]
I've searched for words 'private' and 'protected' on this page but found nothing: https://mobx.js.org/observable-state.html#makeobservable

**Reply by mweststrate** [reply]
Correct. Will update

**Reply by phillipwinder** [reply]
Is there no other way of doing this? 

Gets kind of annoying when there are more than 2 or 3 private variables/methods.

**Reply by kubk** [reply]
@PhillipWinder either use makeAutoObservable instead of makeObservable or use functions instead of classes:
```js
export const createTestStore = () => {
  // non-observable
  const privateVar = ...
  const privateFunction = ...

  // observable
  return makeAutoObservable({
     ...
  })
}

**Reply by joswarmer** [reply]
Took me some time to understand what `MyStore`  was on  [ https://mobx.js.org/observable-state.html#makeobservable](https://mobx.js.org/observable-state.html#makeobservable) but that seems to be the current class.
